### ISSUE TITLE
[BLOCKED] Task: Add a document builder

### ISSUE THREAD

**Issue by ReinderVosDeWael** [childmindresearch/agate#70]
### Description

Across many tasks at CMI, we have a need to write documents based on several other documents. Some examples are: an aims page for a grant proposal or a PRD for a product. The goal of this task is to build this document builder.

### Tasks

- [ ] Build a new frontend page `/document-builder`
- [ ] The page should allow for selecting pre-set documents, or building your own.
  - [ ] For the first version, the PRD and Aims pre-sets should be built (see notes below).  
  - [ ] Building your own comes with the following customizations for the first version:
    - [ ] Custom Headings up to a depth of two.
    - [ ] Custom instructions on a per-paragraph level.
  - [ ] Saving customizations is out-of-scope. 
- [ ] The user should be able to provide further full-document instructions.
- [ ] The user should be able to upload up to five files with a description of each file. 

### Freeform Notes

PRD: https://docs.google.com/document/d/1x7rXMu4wt3f--nQ2dm4IWz7qOKyk2zM68vPORJpSJw0/edit
Aims: https://docs.google.com/document/d/15CKCt6Qgxag9crTlCTsCU2-DnCmh1irgvDyFnGulfVQ/edit
Aims 2: https://docs.google.com/document/d/15c6VUKljM7WvjGUOwCkOEx6srbin2iTaSaJ6JPEn7vg/edit#heading=h.239ncq5i6qmr

**Comment by ReinderVosDeWael** [comment]
Task is a candidate for splitting up into a PRD and Aims task due to its large size.

**Comment by ReinderVosDeWael** [comment]
Blocked on review of output quality before continuing.
